### PR TITLE
chore:  change the klona strategy of setting of values to evalTree and evalActionBindings

### PR DESCRIPTION
**Pull request by vsvamsi1**
## Description
Changed the klona strategy of evalActionBindings and parsed values within evaluateTree to use klonaJSON instead of klona. Seeing a 10-20% reduction in webworker scripting and a 4 second drop in LCP within a windows machine.

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12275133735>
> Commit: dbe08f90ef23d24131f27b7731c953a584fdd61f
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12275133735&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Wed, 11 Dec 2024 12:37:03 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling during data evaluation, providing better context for debugging.
	- Improved logic for evaluating dynamic properties within the data tree.

- **Bug Fixes**
	- Addressed issues with data cloning methods to ensure more reliable data handling.

- **Refactor**
	- Updated method signatures for improved clarity and consistency across the evaluation processes.

### CODE REVIEW

**Review by rishabhrathod01 — DISMISSED** [review]
LGTM

File: app/client/src/workers/common/DataTreeEvaluator/index.ts [hunk 1]
```diff
@@ -1290,7 +1290,7 @@ export default class DataTreeEvaluator {
              * Their evaluated values need to be reset only when the variable is modified by the user.
              * When uneval value of a js variable hasn't changed, it means that the previously evaluated values are in both trees already  */
             if (!skipVariableValueAssignment) {
-              const valueForSafeTree = klona(evalValue);
+              const valueForSafeTree = klonaJSON(evalValue);
```
Comment (dvj1988) [inline app/client/src/workers/common/DataTreeEvaluator/index.ts:1293]: @rishabhrathod01 What are the allowed dataTypes for jsobject variables? We need to check if all these datatypes are supported by `klonaJSON`.

Comment (rishabhrathod01) [reply app/client/src/workers/common/DataTreeEvaluator/index.ts:1293]: string, boolean, number, object, array, map and set are the main dataTypes that are supposed to be supported.

Comment (rishabhrathod01) [reply app/client/src/workers/common/DataTreeEvaluator/index.ts:1293]: Can we please test for map, set and date object @vsvamsi1

Comment (vsvamsi1) [reply app/client/src/workers/common/DataTreeEvaluator/index.ts:1293]: Sure can you suggest some test scenarios for this?

Comment (dvj1988) [reply app/client/src/workers/common/DataTreeEvaluator/index.ts:1293]: Create all types variables in jsobject and try to read and update them through the app. Check if you see anomalies.

**Review by dvj1988 — APPROVED** [review]

**Pull request merged**